consensus/polybft/signer/private.go

```diff
@@ -4,50 +4,56 @@ import (
 	"errors"
 	"math/big"
 
-	"github.com/kilic/bn254"
+	bn256 "github.com/umbracle/go-eth-bn256"
 )
 
+// PrivateKey holds private key for bls implementation
 type PrivateKey struct {
-	p *big.Int
+	s *big.Int
 }
 
 // PublicKey returns the public key from the PrivateKey
 func (p *PrivateKey) PublicKey() *PublicKey {
-	g2 := bn254.NewG2()
-	public := g2.New()
+	g2 := new(bn256.G2)
+	g2 = g2.ScalarMult(g2Point, p.s)
 
-	g2.MulScalar(public, g2.One(), p.p)
+	return &PublicKey{g2: g2}
+}
+
+// MarshalJSON marshal the key to json bytes.
+func (p *PrivateKey) MarshalJSON() ([]byte, error) {
+	if p.s == nil {
```

Comment:
If this happens, someone is not using correctly the library and it should fail during tests.

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1161/commits/c39a660fed18b2128cee9bc0329b13d278d6c8b5